codeguru_profiler_agent/profiler_runner.py

```diff
@@ -51,7 +51,7 @@ def start(self):
 
         :return: True if the profiler was started successfully; False otherwise.
         """
-        if self.profiler_disabler.should_stop_profiling():
+        if self.profiler_disabler.should_stop_sampling():
             logger.info("Profiler will not start.")
```

Comment:
Could you update the logger with details that will not start due to sampling's conditions? Same above at "if self.profiler_disabler.should_stop_profiling(profile=self.collector.profile)", add a log similar, to make a clear distinction between the 2 when debugging.

Reply:
I agree; but I dont think I would update the log here.

I think it would fits better if I update the log message in ProfilerDisabler such as: 

"Profiler sampling cpu usage limit reached: {:.2f} % (limit: {:.2f} %), will stop CodeGuru Profiler." vs                 "Profiler overall cpu usage limit reached: {:.2f} % (limit: {:.2f} %), will stop CodeGuru Profiler."

For exactly this line, the only way that disabler returned False for should_stop_profiling is when killswitch file is detected (cpu and memory check should not be carried out as the agent just started and timer would not do the check until we have several entries in the timer. 

After all, if disabler returns false here, it must return error message with either killswitch is detected or cpu/ memory limit breached (either related to sampling or overall). :D